core/tests/test_utils.py

```diff
@@ -1809,12 +1812,15 @@ def swap(self, obj, attr, newvalue):
         the generator will immediately raise StopIteration, and contextlib will
         raise a RuntimeError.
         """
-        original = getattr(obj, attr)
+        original = getattr(obj, attr, NO_ATTRIBUTE)
```

Comment:
I'm not really keen on this change. Conceptually, "swap" and "add/delete" are different concepts. If we really need add/delete, we should use a different contextmanager (but it would be better if we didn't need add/delete).

Why is this needed? In general, I think we should try to keep the changes in this PR minimal and incremental to avoid breaking other machines/workflows (and it's also good practice anyway).